zio-http/src/main/scala/zio/http/Http.scala

```diff
@@ -721,7 +722,8 @@ object Http {
   /**
    * Provides access to the request's ChannelHandlerContext
    */
-  def context: Http[Any, Nothing, Request, ChannelHandlerContext] = Http.fromFunction[Request](_.unsafeContext)
+  def context: Http[Any, Nothing, Request, ChannelHandlerContext] =
+    Http.fromFunction[Request](request => Unsafe.unsafe { implicit u => request.unsafe.context })
```

Comment:
This should probably be `fromFunctionZIO` to handle the unsafety

Reply:
Yes although it will have the same effect (die) unless we want to expose an error here.